File: weather-demo-app-apple/Weather/UI/Current/CurrentWeatherScreen.swift
//
//  CurrentWeatherScreen.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

import SwiftUI

struct CurrentWeatherScreen: View {
    @ObservedObject var viewModel: CurrentWeatherViewModel
    let onOpenForecast: (String) -> Void

    var body: some View {
        CurrentWeatherView(
            city: viewModel.city,
            state: viewModel.state,
            onCityChanged: viewModel.onCityChanged,
            onSearchClicked: {
                Task {
                    await viewModel.search(city: viewModel.city)
                }
            },
            onRetryClicked: {
                Task {
                    await viewModel.retry()
                }
            },
            onForecastClicked: {
                let forecastCity: String
                switch viewModel.state {
                case .success(let weather):
                    forecastCity = weather.city
                default:
                    forecastCity = viewModel.city
                }
                onOpenForecast(forecastCity)
            }
        )
        .navigationTitle("Current Weather")
        .navigationBarTitleDisplayMode(.inline)
        .task {
            await viewModel.loadLastCity()
        }
    }
}

File: weather-demo-app-android/app/src/main/java/com/demo/weather/ui/current/CurrentWeatherScreen.kt
package com.demo.weather.ui.current

import androidx.compose.foundation.layout.padding
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CurrentWeatherScreen(
    viewModel: CurrentWeatherViewModel,
    onOpenForecast: (String) -> Unit
) {
    val state by viewModel.state.collectAsState()
    val city by viewModel.city.collectAsState()

    LaunchedEffect(Unit) {
        viewModel.loadLastCity()
    }

    Scaffold(
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Current Weather", fontWeight = FontWeight.Bold) }
            )
        }
    ) { innerPadding ->

        CurrentWeatherView(
            city = city,
            state = state,
            onCityChanged = viewModel::onCityChanged,
            onSearchClicked = { viewModel.search(city) },
            onRetryClicked = { viewModel.retry() },
            onForecastClicked = {
                (state as? CurrentWeatherViewState.Success)?.let {
                    onOpenForecast(it.weather.city)
                }
            },
            modifier = Modifier.padding(innerPadding)
        )
    }
}